### PR TITLE
own and inherited attributes display functionality in docgen

### PR DESCRIPTION
**Pull request by sujaypatil96**
This PR addresses issue #878

### CODE REVIEW

File: linkml/generators/docgen.py [hunk 1]
```diff
@@ -668,6 +668,37 @@ def _is_single_file_format(self, format: str):
...
 
+    def fetch_own_attributes_of_class(self, cls: ClassDefinition) -> List[SlotDefinition]:
+        """Fetch list of all own attributes of a class, i.e., 
+        all slots that belong to the domain of a class.
+
+        :param cls: class for which we want to determine the attributes
+        :return: list of all own attributes of a class
+        """
+        sv = self.schemaview
+
```
Comment (cmungall) [inline linkml/generators/docgen.py:679]: remove blank lines

File: tests/test_generators/input/kitchen_sink.yaml [hunk 2]
```diff
@@ -276,32 +276,38 @@ classes:
 
 slots:
   employed at:
+    domain_of: Person
```
Comment (cmungall) [inline tests/test_generators/input/kitchen_sink.yaml:279]: remove these. These are not asserted

**Review by cmungall — CHANGES_REQUESTED** [review]
for this to be useful, it shouldn't rely on domain_of being asserted

**Comment by sujaypatil96** [comment]
So we need to write logic that computes all classes that are in the domain of a slot without the explicit assertion of domain_of?

File: linkml/generators/docgen.py [hunk 3]
```diff
@@ -546,6 +546,43 @@ def _is_single_file_format(self, format: str):
         else:
             return False
 
+    def fetch_own_attributes_of_class(self, cls: ClassDefinition) -> List[SlotDefinitionName]:
```
Comment (cmungall) [inline linkml/generators/docgen.py:549]: call this `get_direct_slots`

File: linkml/generators/docgen.py [hunk 4]
```diff
@@ -546,6 +546,43 @@ def _is_single_file_format(self, format: str):
...
+        """Fetch list of all own attributes of a class, i.e., 
+        all slots that belong to the domain of a class.
+
+        :param cls: class for which we want to determine the attributes
+        :return: list of all own attributes of a class
+        """
+        return cls.slots + list(cls.attributes.keys())
+
+    def fetch_inherited_attributes_of_class(self, cls: ClassDefinition) -> List[SlotDefinitionName]:
```
Comment (cmungall) [inline linkml/generators/docgen.py:558]: call this `get_indirect_slots`

File: linkml/generators/docgen.py [hunk 5]
```diff
@@ -546,6 +546,43 @@ def _is_single_file_format(self, format: str):
...
+        :param cls: class for which we want to determine the attributes
+        :return: list of all own attributes of a class
+        """
+        sv = self.schemaview
+        
+        slot_list = [slot.name for slot in sv.class_induced_slots(class_name=cls.name)]
+
+        return list(set(slot_list).difference(self.fetch_own_attributes_of_class(cls)))
+
+    def fetch_mixed_in_slots_of_class(self, cls: ClassDefinition) -> Dict[str, List[str]]:
```
Comment (cmungall) [inline linkml/generators/docgen.py:571]: call this `get_mixin_inherited_slots`

**Pull request merged**